sqlmesh/dbt/target.py

```diff
@@ -255,3 +265,71 @@ def to_sqlmesh(self) -> ConnectionConfig:
             access_token=self.token,
             concurrent_tasks=self.threads,
         )
+
+
+class BigQueryConfig(TargetConfig):
+    """
+    Project connection and operational configuration for the BigQuery target
+
+    Args:
+        schema_: Overrides
+        type: The type of the target (bigquery)
+        method: The BigQuery authentication method to use
+        project: The BigQuery project to connect to
+        location: The BigQuery location to connect to
+        keyfile: The path to the BigQuery keyfile
+        keyfile_json: The BigQuery keyfile as a JSON string
+        token: The BigQuery token
+        refresh_token: The BigQuery refresh token
+        client_id: The BigQuery client ID
+        client_secret: The BigQuery client secret
+        token_uri: The BigQuery token URI
+        scopes: The BigQuery scopes
+    """
+
+    type: Literal["bigquery"] = "bigquery"
+    method: t.Optional[str] = BigQueryConnectionMethod.OAUTH
+    schema_: t.Optional[str] = Field(None, alias="schema")  # type: ignore
```

Comment:
schema is inherited from the base class, so I don't think you need to define it here too.

Reply:
I need it to be optional.

Reply:
schema is used within the model's fully qualified name, so it must be set. For BigQuery, we'll need to figure out how to provide schema when dataset is used instead.

Reply:
This does that: https://github.com/TobikoData/sqlmesh/blob/5ec62b767ab66bec9ffb6348a74626b45c2e3322/sqlmesh/dbt/target.py#L309-L316

So `schema` is optional but it makes sure that either `schema` or `dataset` are set.